packages/headless/doc-parser/src/interface-resolver.ts

```diff
@@ -62,12 +62,36 @@ function filterOverridesAndCombine(
   return members.concat(filtered);
 }
 
+function sortEntities(entities: AnyEntity[]) {
+  const kindOrder: Record<EntityKind, number> = {
+    primitive: 0,
+    'primitive-with-type-alias': 0,
+    object: 0,
+    function: 1,
+  };
```

Comment:
Once concern with the map is if we add another kind of entity, we would need to come back and maintain it. Couldn't we instead filter and keep everything that is not a function, and sort those?

```
const funcEntities = entities.filter(isFunctionEntity)
const nonFuncEntities = entities.filter(entity => !isFunctionEntity(entity))

// sort

return nonFuncEtities.concat(funcEntities)
```